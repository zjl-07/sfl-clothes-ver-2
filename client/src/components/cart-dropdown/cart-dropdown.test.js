import React from "react";
import { shallow } from "enzyme";
import CartDropdown from "./cart-dropdown.jsx";
import CartItem from "components/cart-item/cart-item";
import { toggleCartHidden } from "redux/cart/cart.action";

describe("Cart Dropdown", () => {
  let wrapper;
  let mockHistory;
  let mockDispatch;
  const mockCartItems = [{ id: 1 }, { id: 2 }, { id: 3 }];

  beforeEach(() => {
    mockHistory = {
      push: jest.fn()
    };

    mockDispatch = jest.fn();

    const mockProps = {
      cartItems: mockCartItems,
      history: mockHistory,
      dispatch: mockDispatch
    };

    wrapper = shallow(<CartDropdown />);
  });

  // it("should render cart dropdown component", () => {
  //   expect(wrapper).toMatchSnapshot();
  // });

  // it("should call history.push when button is clicked", () => {
  //   // wrapper.find("DropdownButton").simulate("click");
  //   // expect(wrapper.find("DropdownButton").length).toEqual(1);
  //   // expect(mockHistory.push).toHaveBeenCalled();
  //   // expect(mockDispatch).toHaveBeenCalledWith(toggleCartHidden());
  // });

  it("should render cart item length equals to mock item length", () => {
    console.log(wrapper);
    expect(wrapper.find("DropdownButton").length).toEqual(1);
  });
});
