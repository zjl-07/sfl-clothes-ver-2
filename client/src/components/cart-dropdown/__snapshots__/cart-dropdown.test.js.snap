// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Dropdown should render cart dropdown component 1`] = `ShallowWrapper {}`;
